clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -176,10 +176,25 @@ public Password getPassword(String key) {
      * @return a Map containing the settings with the prefix
      */
     public Map<String, Object> originalsWithPrefix(String prefix) {
+        return originalsWithPrefix(prefix, true);
+    }
+
+    /**
+     * Gets all original settings with the given prefix.
+     *
+     * @param prefix the prefix to use as a filter
+     * @param strip strip the prefix before adding to the output if set true
+     * @return a Map containing the settings with the prefix
+     */
+    public Map<String, Object> originalsWithPrefix(String prefix, boolean strip) {
         Map<String, Object> result = new RecordingMap<>(prefix, false);
         for (Map.Entry<String, ?> entry : originals.entrySet()) {
-            if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length())
-                result.put(entry.getKey().substring(prefix.length()), entry.getValue());
+            if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length()) {
+                if (strip)
```

Comment:
It seems this check can be lifted above the check at line 192.
Since we don't need to check prefix if strip == false.

Reply:
Actually we still need to check prefix even if strip == false, the only difference is that if the check of the prefix succeeds, whether or not to maintain the prefix in the returned map.